YoCode/GitCheck.cs

```diff
@@ -0,0 +1,146 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Diagnostics;
+using System.IO;
+
+namespace YoCode
+{
+    public class GitCheck
+    {
+        public string GIT_INSTALL_DIRECTORY { get; set; } = @"C:\Program Files\Git";
+        //public string REPOSITORY_PATH = @"..\..\..\..\..\YoCode";
+        public string REPOSITORY_PATH = @"C:\Users\ukmzil\source\repos\Tests Sent by People\Real\simon-jones";
+
+        string output { get; set; }
+        string lastAuthor { get; set; }
+        bool result { get; set; }
+
+
+        public List<string> hostDomains = new List<string>();
+
+
+        public bool ExecuteTheCheck()
+        {
+            var p = new Process();
+            return ExecuteTheCheck(p);
+        }
+
+        public bool ExecuteTheCheck(Process p)
+        {
+            p.StartInfo = SetProcessStartInfo(REPOSITORY_PATH);
+            p.Start();
+
+            output = p.StandardOutput.ReadToEnd();
+            lastAuthor = getLastAuthor(output);
+            return GitHasBeenUsed(lastAuthor,hostDomains);
+        }
+
+        public ProcessStartInfo SetProcessStartInfo(String PATH)
+        {
+            var psi = new ProcessStartInfo();
+            psi.CreateNoWindow = true;
+            psi.RedirectStandardError = true;
+            psi.RedirectStandardOutput = true;
+            psi.WorkingDirectory = PATH;
+            psi.FileName = GIT_INSTALL_DIRECTORY + @"\bin\git.exe";
```

Comment:
Like I mentioned in standup, I think it's okay to just say "git" here without knowing the whole path, because if they have it installed that just work, and it avoids you having to hardcode the path, which might break if they installed it somewhere else.

Reply:
Fixed